frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchAction.java

```diff
@@ -95,4 +104,21 @@ private void doPull(Project project, GitRepository gitRepository, String branchN
     new FetchBackgroundable(project, gitRepository, refspecRemoteLocal, GitRemote.DOT,
         /* taskTitle */ GitMacheteBundle.message("action.GitMachete.BasePullBranchAction.task-title")).queue();
   }
+
+  private void doPull(Project project, GitRepository gitRepository, String branchName) {
+    var firstRemote = Option.ofOptional(gitRepository.getRemotes().stream().findFirst()).getOrNull();
```

Comment:
Whoops... any chance this remote could be extracted from branch tracking info?

Reply:
Yes, pulled branch can have other remote than the first remote from collection of repository remotes.
`remoteName` field can be added to `GitMacheteRemoteBranch` (like is in `GitCoreRemoteBranchSnapshot`), then small revamp of `GitMacheteRepository#getRemoteTrackingBranchForCoreLocalBranch` in line 421 is needed to get remote name from `coreRemoteBranch` and pass it to `GitMacheteRemoteBranch` constructor. And then in `BasePullBranchAction#doPull` if we pass `GitMacheteRemoteBranch`that is remote tracking branch of our current branch to it, we can choose proper `GitRemote` (e.g. `gitRepository.getRemotes().stream().filter(remote -> remote.getName().equals(remoteName)).findFirst()`.

Reply:
accessed like `doFetch` and extracted